lib/rspec/support/method_signature_verifier.rb

```diff
@@ -357,7 +361,14 @@ def unlimited_args?
 
       def split_args(*args)
         kw_args = if @signature.has_kw_args_in?(args)
-                    args.pop.keys
+                    last = args.pop
+                    non_kw_args = last.reject { |k, _| k.is_a?(Symbol) }
+                    if non_kw_args.empty?
+                      last.keys
+                    else
+                      args << non_kw_args
+                      last.select { |k, _| k.is_a?(Symbol) }.keys
```

Comment:
Are you saying with this that Ruby will mix keyword args and hash keys?

Reply:
If you provide an anonymous Hash as the last argument only the symbols from that argument will be treated as keyword arguments, the rest will be interpreted as positional one.

```
def foo(x = {}, y:); puts "#{x}, #{y}"; end

foo 'a' => 1, y: 2
=> {"a"=>1}, 2
foo x: 3, y: 2
=> ArgumentError (unknown keyword: x)

def bar(x:, y:); end

bar 'x' => 1
=> ArgumentError (wrong number of arguments (given 1, expected 0; required keywords: x, y))
bar x: 1, y: 2, z: 3
=> ArgumentError (unknown keyword: z)
```

I'm trying to cover the behavior in [this spec](https://github.com/rspec/rspec-support/blob/778f143da167ed4f3f8e8715543ddec153694aaa/spec/rspec/support/method_signature_verifier_spec.rb#L387)